posts/2020-11-20-JNDI-gRPC.adoc

```diff
@@ -0,0 +1,361 @@
+
+These two features which were previously in our beta are now available in this months GA: `grpc-1.0`, which enables gRPC services, and `grpcClient-1.0`, which enables the use of a gRPC client for outbound calls.
+
+The `grpc-1.0` feature works by scanning web apps for gRPC service implementations, through implementors of `io.grpc.BindableService`. The web app must include the protocol buffer compiler-generated code for the services it intends to provide, and additionally the service class must provide a no-argument constructor. The web app does not need to include any core gRPC libraries; those are provided by the Liberty runtime. Once a gRPC service is scanned and started, it becomes accessible to remote gRPC clients on the configured HTTP ports.
+
+The `grpcClient-1.0` feature provides applications with access to a link:https://netty.io/[Netty] gRPC client, as well as the related libraries. A web app must provide a client implementation and stubs, and can make outbound calls with a `io.grpc.ManagedChannel` without needing to provide the supporting client libraries.
+
+You can now sample gRPC by either following the instructions in this link:https://github.com/OpenLiberty/sample-grpc[GitHub repositories] README or following this basic Hello World service (add the  `grpc-1.0` to the `server.xml`):
+
+[source, java]
+----
+package com.ibm.ws.grpc;
+
+import com.ibm.ws.grpc.beans.GreetingBean;
+
+import io.grpc.examples.helloworld.GreeterGrpc;
+import io.grpc.examples.helloworld.HelloReply;
+import io.grpc.examples.helloworld.HelloRequest;
+import io.grpc.stub.StreamObserver;
+
+public class HelloWorldService extends GreeterGrpc.GreeterImplBase {
+
+    public HelloWorldService(){}
+
+    @Override
+    public void sayHello(HelloRequest req, StreamObserver<HelloReply> responseObserver) {
+        HelloReply reply = HelloReply.newBuilder().setMessage("Hello " + req.getName()).build();
+        responseObserver.onNext(reply);
+        responseObserver.onCompleted();
+    }
+}
+----
+
+For this example, the application must provide the link:https://github.com/grpc/grpc-java/blob/master/examples/src/main/proto/helloworld.proto[helloworld protof definition] along with the protobuf compiler output. No additional libraries need to be provided with the application, and once it's started the helloworld greeter service will be accessible on the server's HTTP endpoints.
+
+For a client example, a basic Servlet using gRPC can be defined via `grpcClient-1.0` with:
```

Comment:
`via grpcClient-1.0` should be `using the grpcClient-1.0 feature`

Reply:
Done